### PR TITLE
Add deleter metrics modeled on archiver metrics [BW-632]

### PR DESCRIPTION
**Pull request by rsasch**
Closes https://broadworkbench.atlassian.net/browse/BW-632

### CODE REVIEW

File: services/src/main/scala/cromwell/services/metadata/impl/deleter/DeleteMetadataActor.scala [hunk 1]
```diff
@@ -29,19 +32,74 @@ class DeleteMetadataActor(deleteMetadataConfig: DeleteMetadataConfig,
   implicit val ec: ExecutionContext = context.dispatcher
   implicit val askTimeout: Timeout = new Timeout(60.seconds)
 
+  private val deleterMetricsBasePath: NonEmptyList[String] = MetadataServiceActor.MetadataInstrumentationPrefix :+ "archiver"
```
Comment (aednichols) [inline services/src/main/scala/cromwell/services/metadata/impl/deleter/DeleteMetadataActor.scala:35]: Copy-paste error?
```suggestion
  private val deleterMetricsBasePath: NonEmptyList[String] = MetadataServiceActor.MetadataInstrumentationPrefix :+ "deleter"
```

Comment (rsasch) [reply services/src/main/scala/cromwell/services/metadata/impl/deleter/DeleteMetadataActor.scala:35]: whoops — thanks for the catch!

File: database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala [hunk 2]
```diff
@@ -125,6 +125,16 @@ trait WorkflowMetadataSummaryEntryComponent {
     } yield summaryEntry
   }
 
+  private def fetchAllWorkflowsToDeleteThatEndedOnOrBeforeThresholdTimestamp(workflowStatuses: List[String],
+                                                                              workflowEndTimestampThreshold: Timestamp): Query[WorkflowMetadataSummaryEntries, WorkflowMetadataSummaryEntry, Seq] = {
+    for {
+      summaryEntry <- workflowMetadataSummaryEntries
+      if summaryEntry.workflowStatus.inSet(workflowStatuses)
```
Comment (salonishah11) [inline database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala:132]: I think when looking for workflows to delete, we might not need to check the workflow status. We only need to check for archive status and end time just like [workflowIdsByArchiveStatusAndEndedOnOrBeforeThresholdTimestamp](https://github.com/broadinstitute/cromwell/blob/187c49defd1ef3c2df4133e6e9b9c92edfce79b4/database/sql/src/main/scala/cromwell/database/slick/tables/WorkflowMetadataSummaryEntryComponent.scala#L74-L81)

**Review by aednichols — APPROVED** [review]

**Review by salonishah11 — CHANGES_REQUESTED** [review]
1 change requested and few minor copy paste errors

**Review by salonishah11 — APPROVED** [review]
Just 1 small nit comment [here](https://github.com/broadinstitute/cromwell/pull/6340/files#r629661412). Otherwise LGTM 👍

**Pull request merged**